compiler+runtime/src/cpp/jank/read/parse.cpp

```diff
@@ -1513,7 +1513,7 @@ namespace jank::read::parse
     auto const token((*token_current).expect_ok());
     ++token_current;
     auto const b(std::get<bool>(token.data));
-    return object_source_info{ make_box<obj::boolean>(b), token, token };
+    return object_source_info{ b ? jank_true : jank_false, token, token };
```

Comment:
An alternative would be to fix this in make_box, but I wasn't sure.

Reply:
Oh, it's because of the `<obj::boolean>`. It doesn't get into the overload you linked, since that overload is not a template. We should fix this in the `make_box` template instead.

It's getting here: https://github.com/jank-lang/jank/blob/main/compiler%2Bruntime/include/cpp/jank/runtime/oref.hpp#L427

Ideally, we can specialize this template where `T::obj_type == object_type::boolean` to just return the constants. Otherwise, we're actually creating new bools which are not going to be identical with the constants. That shouldn't be possible.